src/metabase/driver/util.clj

```diff
@@ -20,6 +20,75 @@
            javax.net.SocketFactory
            [javax.net.ssl SSLContext TrustManagerFactory X509TrustManager]))
 
+(def ^:private connection-error-messages
+  "Generic error messages that drivers should return in their implementation
+  of [[metabase.driver/humanize-connection-error-message]]."
+  {:cannot-connect-check-host-and-port
+   {:message [(deferred-tru "Hmm, we couldn''t connect to the database.")
+              " "
+              (deferred-tru "Make sure your Host and Port settings are correct")]
+    :errors  {:host (deferred-tru "check your host settings")
+              :port (deferred-tru "check your port settings")}}
+
+   :ssh-tunnel-auth-fail
+   {:message [(deferred-tru "We couldn''t connect to the SSH tunnel host.")
+              " "
+              (deferred-tru "Check the Username and Password.")]
+    :errors  {:tunnel-user (deferred-tru "check your username")
+              :tunnel-pass (deferred-tru "check your password")}}
+
+   :ssh-tunnel-connection-fail
+   {:message [(deferred-tru "We couldn''t connect to the SSH tunnel host.")
+              " "
+              (deferred-tru "Check the Host and Port.")]
+    :errors  {:tunnel-host (deferred-tru "check your host settings")
+              :tunnel-port (deferred-tru "check your port settings")}}
+
+   :database-name-incorrect
+   {:message (deferred-tru "Looks like the Database name is incorrect.")
+    :errors  {:dbname (deferred-tru "check your database name settings")}}
+
+   :invalid-hostname
+   {:message [(deferred-tru "It looks like your Host is invalid.")
+              " "
+              (deferred-tru "Please double-check it and try again.")]
+    :errors  {:host (deferred-tru "check your host settings")}}
+
+   :password-incorrect
+   {:message (deferred-tru "Looks like your Password is incorrect.")
+    :errors  {:password (deferred-tru "check your password")}}
+
+   :password-required
+   {:message (deferred-tru "Looks like you forgot to enter your Password.")
+    :errors  {:password (deferred-tru "check your password")}}
+
+   :username-incorrect
+   {:message (deferred-tru "Looks like your Username is incorrect.")
+    :errors  {:user (deferred-tru "check your username")}}
+
+   :username-or-password-incorrect
+   {:message (deferred-tru "Looks like the Username or Password is incorrect.")
+    :errors  {:user     (deferred-tru "check your username")
+              :password (deferred-tru "check your password")}}
+
+   :certificate-not-trusted
+   {:message (deferred-tru "Server certificate not trusted - did you specify the correct SSL certificate chain?")}
+
+   :requires-ssl
+   {:message (deferred-tru "Server appears to require SSL - please enable SSL below")
+    :errors  {:ssl (deferred-tru "please enable SSL")}}})
+
+(defn- force-tr [text-or-vector]
```

Comment:
This `force-tr` stuff is clever but I think we need a more generic fix for this issue. Just filed https://github.com/metabase/metabase/issues/22897. This looks fine for now.

Reply:
Nvm, https://github.com/metabase/metabase/issues/10770 was already open for this